File: Cuarto/4.Ejercicio.py
# -*- coding: utf-8 -*-
"""
@author: NOEMI
"""

def calculadora(funcion, a, b):
    funcion(a,b)
    
def suma(x, y):
    print("Suma:",x+y)   
def resta(x, y):
    print("Resta:",x-y)   
def multiplicacion(x, y):
    print("Multiplicacion",x*y)   
def division(x, y):
    print("Division:",x/y)


a = int(input("Primer Numero Entrada: "))
b = int(input("Segundo Numero Entrada: "))
calculadora(suma,a,b)
calculadora(resta,a,b)
calculadora(multiplicacion,a,b)
calculadora(division,a,b)

File: Cuarto/4.Ejercicio.scala
object Cuarto {
    def calculadora(funcion: (Int, Int) => Int, a: Int, b: Int) = {
        funcion(a,b)
    }
    
    def suma(x: Int, y: Int): Int = {
        return x + y
    }
    def resta(x: Int, y: Int) : Int = {
       return x - y
    }
    def multiplicacion(x: Int, y: Int) : Int = {
        return x * y
    }
    def division(x: Int, y: Int) : Int = {
        return x / y
    }

    def main(args: Array[String]) {
        print("Primero numero: ")
        var a = scala.io.StdIn.readInt()
        print("Segundo numero: ")
        var b = scala.io.StdIn.readInt()
        println("Suma: " + calculadora(suma,a,b));
        println("Resta: " + calculadora(resta,a,b));
        println("Multiplicacion: " + calculadora(multiplicacion,a,b));
        println("Division: " + calculadora(division,a,b));
    }
}